### PR TITLE
chore: sourcemod architecture and CI

### PR DESCRIPTION
**Pull request by Rushaway**

### CODE REVIEW

File: sourceknight.ymal [hunk 1]
```diff
@@ -0,0 +1,23 @@
+project:
```
Comment (maximee-leroyy) [inline sourceknight.ymal]: sourceknight.yaml

**Comment by maximee-leroyy** [comment]
could you separate the architecture change in another MR so that we have a proper diff for autosilencer.sp

**Comment by Rushaway** [comment]
Sure, I made CI + SM architecture in this PR.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      - uses: actions/checkout@v4
+      - name: Build sourcemod plugin
+        uses: maxime1907/action-sourceknight@v1
+        with:
+          cmd: build
+
+      - name: Create package
+        run: |
+          mkdir -p /tmp/package
+          cp -R {.sourceknight/package/*,common} /tmp/package
```
Comment (maximee-leroyy) [inline .github/workflows/ci.yml:25]: no common folder here, remove

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+          cp -R addons/sourcemod/translations /tmp/package/common/addons/sourcemod/
```
Comment (maximee-leroyy) [inline .github/workflows/ci.yml:26]: no translations also for now

**Pull request merged**